File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        maxCount = 1
        currentCount = 1
        uniqueNums = set(nums)
        sortedNums = sorted(uniqueNums)
        for i in range(1, len(sortedNums)):
            if sortedNums[i] == sortedNums[i-1] + 1:
                currentCount += 1
            else:
                maxCount = max(currentCount, maxCount)
                currentCount = 1
        return max(currentCount, maxCount)

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
var longestConsecutive = function(nums) {
    if(nums.length===0) return 0;
    nums.sort((a,b)=>a-b)
    let uniqueNums = new Set(nums);
    let num = Array.from(uniqueNums)
    let longestLength = 1;
    let currentLength = 1;
    for(let index = 1 ; index<num.length ;index++){
        if(num[index] === num[index-1]+1){
            currentLength++;
            longestLength = Math.max(currentLength,longestLength);
        } else {
            longestLength = Math.max(currentLength,longestLength);
            currentLength = 1;
        }
    }
    return longestLength;
};